packages/devtools/src/sourceMap/loadOriginalSourceLoc.ts

```
@@ -0,0 +1,178 @@
+    }
+
+    sourceMappingURLMatch = sourceMapRegex.exec(runtimeSourceCode);
+  }
+
+  // No inline sourcemap found. Check if there's external source map
+  let sourceMappingURL = externalSourceMapURLs.pop();
+  if (!sourceMappingURL) {
+    throw new Error(
+      `[Griffel devtools] extractAndLoadSourceMapJSON() encountered string "sourceMappingURL" in file ${source} that does not contain a valid source map`,
+    );
+  }
+
+  if (externalSourceMapURLs.length > 0) {
+    console.warn(
+      `[Griffel devtools] extractAndLoadSourceMapJSON() More than one external source map detected in the file ${source}:`,
+    );
+    externalSourceMapURLs.forEach(sourceMappingURL => {
+      console.warn(`[Griffel devtools] extractAndLoadSourceMapJSON() skipping source map "${sourceMappingURL}"`);
+    });
+  }
+
+  if (!sourceMappingURL.startsWith('http') && !sourceMappingURL.startsWith('/')) {
+    const lastSlashIdx = source.lastIndexOf('/');
+    if (lastSlashIdx !== -1) {
+      const baseURL = source.slice(0, lastSlashIdx);
+      sourceMappingURL = `${baseURL}/${sourceMappingURL}`;
+    }
+  }
+
+  const sourceMapContent = await fetchFiles(sourceMappingURL);
+  const sourceMapJSON = JSON.parse(sourceMapContent);
+  sourceMapJSONs.set(source, sourceMapJSON);
+
+  return getOriginalPosition(sourceMapJSON, sourceLoc);
+}
+
+async function fetchFiles(url: string): Promise<string> {
+  try {
+    const response = await fetch(url);
+    if (response.ok) {
+      return response.text();
+    }
+    throw new Error(`[Griffel devtools] fetchRuntimeSource() bad response fetching ${url}: ${response.status}`);
+  } catch (error) {
+    // try to get url content from chrome resources api
+    const resource = (await resources).find(resource => resource.url === url);
+    if (resource) {
+      return new Promise(resolve => resource.getContent(content => resolve(content)));
+    }
+    throw new Error(`[Griffel devtools] fetchFiles() unable to fetch ${url}: ${error}`);
```

```suggestion
```

You never need this error right ? you just want to rethrow the earlier error. If the chrome resource api fetch fails, it will throw a compeltely different error

---

Hmm I didn't get it? I think I need it: if fetch fails, it will be catched, and then I will re-throw it here

---

I made a type in the original suggestion, you should just rethrow the caught error:

If original fetch throws an error and you try to use `resource.getContent`

If `resource.getContent` throws then a different error is thrown.

So there is no reason to rethrow a new error from the `catch` block

---

Oh gotcha!